### ISSUE TITLE
Some difference on benchmark score with lm-eval compared with the original paper

### ISSUE THREAD

**Issue by sjlcwn** [RAIVNLab/MatFormer-OLMo#1]
Hello. I was doing some test on the "MatFormer-OLMo-1300M" checkpoint in this repo. When I measured the performance of some benchmarks and compared them with the data from the paper(Table 14), I found that some benchmark have quite different results.
![Result](https://github.com/RAIVNLab/MatFormer-OLMo/assets/24711960/1928cca3-b0e3-4c70-856d-d9f7655643ce)
Generally, the results I get from MatFormer-OLMo-1300M with [lm-eval](https://github.com/EleutherAI/lm-evaluation-harness) has lower accuracy in most benchmarks.

Here is the code I used to collect data and produce the image: [CollectData.zip](https://github.com/RAIVNLab/MatFormer-OLMo/files/14776687/CollectData.zip)
To use the code, run `python collect_data.py` with environment variable MATFORMER_OLMO_CHECKPOINT_PATH set to the checkpoint directory to collect data. Then run `python draw_matformer.py .` to produce the image.
In the code, I packed the model into huggingface PreTrainedModel, in file "utils/LoadModel.py", to use the lm-eval interface .

Any ideas about why the difference happened?

**Comment by adityakusupati** [comment]
Yes, because the paper's models are not OLMO or the same data we used. MatFormer-OLMO is an attempt at open-source reproduction but not equivalent. It is to give an idea of scaling to the public in an Open Source way.

Apologies. The only models from paper which are truly public and match numbers are here (MatViT models): https://github.com/google-research/scenic/tree/main/scenic/projects/matvit

let me know if I could help you with anything.

**Comment by sjlcwn** [comment]
So you think the difference in model structure and training procedure caused the different testing results? I suspected that, but didn't realize the gap would be so large(seems OLMo version almost get lower score in every benchmark, and with large gap in some cases).

If convenient, do you see any mistake or misunderstanding in my testing code that could make the result unreliable?
For example, I set `tokenizer.eos_token_id=0` in collect_data.py to make autoregressive tasks to run, but why it wasn't set when the tokenizer was initially loaded? And would extracting `base_tokenizer` from the originally loaded version in LoadModel.py affect the encoding results in the benchmark?

Thanks
